src/rules/interfaceNameRule.ts

```diff
@@ -74,6 +74,5 @@ function walk(ctx: Lint.WalkContext<{ never: boolean }>): void {
 }
 
 function hasPrefixI(name: string): boolean {
-    // Allow IndexedDB interfaces
-    return name.length >= 2 && name[0] === "I" && isUpperCase(name[1]) && !name.startsWith("IDB");
+    return name.length >= 3 && name[0] === "I" && !isLowerCase(name[1]) && !isUpperCase(name[2]);
```

Comment:
can you add some test cases in `always-prefix/test.ts.lint` as well? I feel like this will affect that option's behavior.

with `always-prefix`, I think the behavior should be:

- failures: `ID`
- valid: `IDB` (or, any all-caps interface name starting with `I`)

I know it feels a little silly since these are edge cases in naming, but we should try to be consistent in some way

Reply:
I can add test yes.
But I didn't change the behaviour for the àlways-prefix` rule.

Actually the test check only if the first letter is `I`.
Now I'm thinking about it, an interface called `Information` should return an error but it doesn't.

With `always-prefix`, I think `ID` should return no error.
You can create an `A` interface, called `IA` with prefix. The same way a `D` interface would be named `ID` with prefix.

To be consistent the best way is to check
`always-prefix && ! hasPrefixI` or `never-prefix && hasPrefixI`